provider/cmd/pulumi-resource-nuage/nuage_provider/bastion.py

```diff
@@ -0,0 +1,131 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Any, Dict, List, Optional
+from dataclasses import dataclass
+
+import pulumi
+import pulumi_tls as tls
+
+import pulumi_aws as aws
+
+
+@dataclass
+class BastionArgs:
+    vpc_id: pulumi.Input[str]
+    vpc_subnets: pulumi.Input[List[str]]
+    ssh_port: pulumi.Input[int]
+
+    @staticmethod
+    def from_inputs(inputs: pulumi.Inputs) -> "BastionArgs":
+        return BastionArgs(
+            vpc_id=inputs["vpcId"],
+            vpc_subnets=inputs["vpcSubnets"],
+            ssh_port=inputs.get("sshPort", 22),
+        )
+
+
+class Bastion(pulumi.ComponentResource):
+    public_ip: pulumi.Output[str]
+    private_key_pem: pulumi.Output[str]
+
+    def __init__(
+        self,
+        name: str,
+        args: BastionArgs,
+        props: Optional[dict] = None,
+        opts: Optional[pulumi.ResourceOptions] = None,
+    ) -> None:
+        super().__init__("nuage:aws:Bastion", name, props, opts)
+
+        private_key = tls.PrivateKey(
+            f"{name}-private-key",
+            algorithm="RSA",
+            rsa_bits=4096,
+            opts=pulumi.ResourceOptions(parent=self),
+        )
+        key_pair = aws.ec2.KeyPair(
+            f"{name}-keypair",
+            public_key=private_key.public_key_openssh,
+            opts=pulumi.ResourceOptions(parent=private_key),
+        )
+
+        security_group = aws.ec2.SecurityGroup(
+            f"{name}/security-group",
+            vpc_id=args.vpc_id,
+            tags={"Name": f"Security group for ssh {name}"},
+            ingress=[
+                aws.ec2.SecurityGroupIngressArgs(
+                    from_port=args.ssh_port,
+                    to_port=args.ssh_port,
+                    protocol="tcp",
+                    cidr_blocks=["0.0.0.0/0"],
+                )
+            ],
+            egress=[
+                aws.ec2.SecurityGroupIngressArgs(
+                    from_port=0,
+                    to_port=0,
+                    protocol="-1",
+                    cidr_blocks=["0.0.0.0/0"],
+                )
+            ],
+            opts=pulumi.ResourceOptions(parent=self),
+        )
+
+        bastion_ami = aws.ec2.get_ami(
+            most_recent=True,
+            filters=[
+                aws.ec2.GetAmiFilterArgs(
+                    name="name",
+                    values=["ubuntu/images/hvm-ssd/ubuntu-focal-20.04-arm64-server-*"],
+                ),
+                aws.ec2.GetAmiFilterArgs(
+                    name="virtualization-type",
+                    values=["hvm"],
+                ),
+            ],
+            owners=["099720109477"],
```

Comment:
Is this an AWS service account of some sort ?

Can you please put a comment if that's the case ?